### DISCUSSION TITLE
Using only the k8s apiserver api/crd is not convenient in some cases for sealos cloud applications requirements

### DISCUSSION THREAD

**Discussion by lingdie** [labring/sealos#2455] [General] [answered]
Using only the k8s apiserver api is not convenient in some cases for sealos cloud applications requirements.

For example, k8s crd can not search items by matename, or conveniently page listed items.

I wonder if we can use k8s crd only as a "database", and rewrite a service layer by using go as backend but not nodejs at client-side? So that we can have a more clearer api for sealos cloud applocations provided by the service layer.

Is there any other ways to extend k8s api?

**Comment by fanux** [comment, marked as answer]
If appstore is an independent APP, I think it can have its own Database, just like laf.
